### PR TITLE
[#3526] feat(trino-connector): Support create Gravitino catalog by using Trino CREATE CATALOG command

### PR DESCRIPTION
**Pull request by diqiu50**
### What changes were proposed in this pull request?

1. Upgrade code to Trino-435
2. Support create Gravitino catalog by CREATE CATALOG statement
3. Disable testers of  Trino Connector 

### Why are the changes needed?

Fix: #3526

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

Exists UT

### CODE REVIEW

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java [hunk 1]
```diff
@@ -420,4 +421,26 @@ public void addMetalake(String metalake) {
...
+
+      GravitinoCatalog catalog = GravitinoCatalog.fromJson(catalogConfig);
+      CatalogConnectorContext.Builder builder =
+          catalogConnectorFactory.createCatalogConnectorContextBuilder(catalog);
+      builder.withMetalake(
+          metalakes.computeIfAbsent(catalog.getMetalake(), this::retrieveMetalake));
+
+      CatalogConnectorContext connectorContext = builder.build();
+      catalogConnectors.put(connectorName, connectorContext);
+      LOG.info("Create connector sucess ");
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java:438]: typo: sucess

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java [hunk 2]
```diff
@@ -420,4 +421,26 @@ public void addMetalake(String metalake) {
   public Set<String> getUsedMetalakes() {
     return usedMetalakes;
   }
+
+  public Connector createConnector(
+      String connectorName, GravitinoConfig config, ConnectorContext context) {
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java:426]: `context` has not been used, do you have other consideration that make it as a parameter?

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java [hunk 3]
```diff
@@ -35,13 +35,15 @@
...
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.MethodOrderer;
 import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestMethodOrder;
 import org.openqa.selenium.By;
 
+@Disabled
```
Comment (mchades) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:46]: The `CatalogPageTest` shouldn't rely on Trino, Can you decouple them?

File: build.gradle.kts [hunk 4]
```diff
@@ -411,12 +407,8 @@ subprojects {
     reports.html.outputLocation.set(file("${rootProject.projectDir}/build/reports/"))
     val skipTests = project.hasProperty("skipTests")
     if (!skipTests) {
-      if (project.name == "trino-connector") {
-        useTestNG()
-        maxHeapSize = "2G"
-      } else {
-        useJUnitPlatform()
-      }
+      maxHeapSize = "2G"
```
Comment (yuqi1129) [inline build.gradle.kts:410]: Is `maxHeapSize` still needed for Junit?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoConfig.java [hunk 5]
```diff
@@ -13,6 +13,8 @@
 public class GravitinoConfig {
 
   public static String GRAVITINO_DYNAMIC_CONNECTOR = "__gravitino.dynamic.connector";
+  public static String GRAVITINO_DYNAMIC_CONNECTOR_CATALOG_CONFIG =
+      "__gravitino.dynamic.connector.catalog.config";
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoConfig.java:17]: Is the value of this key set automatically determined by Trino?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoConfig.java [hunk 6]
```diff
@@ -13,6 +13,8 @@
 public class GravitinoConfig {
 
   public static String GRAVITINO_DYNAMIC_CONNECTOR = "__gravitino.dynamic.connector";
+  public static String GRAVITINO_DYNAMIC_CONNECTOR_CATALOG_CONFIG =
```
Comment (mchades) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoConfig.java:16]: missing `final`?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoMetadata.java [hunk 7]
```diff
@@ -174,9 +175,9 @@ public ColumnMetadata getColumnMetadata(
 
   @Override
   public void createTable(
-      ConnectorSession session, ConnectorTableMetadata tableMetadata, boolean ignoreExisting) {
+      ConnectorSession session, ConnectorTableMetadata tableMetadata, SaveMode saveMode) {
     GravitinoTable table = metadataAdapter.createTable(tableMetadata);
-    catalogConnectorMetadata.createTable(table);
+    catalogConnectorMetadata.createTable(table, saveMode == SaveMode.IGNORE);
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoMetadata.java:180]: What's the meaning of `SaveMode.IGNORE`?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorAdapter.java [hunk 8]
```diff
@@ -22,7 +23,10 @@ default List<PropertyMetadata<?>> getTableProperties() {
   }
 
   /** @return Return internal connector config with Trino. */
-  Map<String, Object> buildInternalConnectorConfig(GravitinoCatalog catalog) throws Exception;
+  Map<String, String> buildInternalConnectorConfig(GravitinoCatalog catalog) throws Exception;
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorAdapter.java:26]: Is this change a result of the interfaced modification introduced by Trino 435?

Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorAdapter.java:26]: It's modified by a newly created connector mechanism.

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java [hunk 9]
```diff
@@ -30,25 +28,22 @@ public HiveConnectorAdapter() {
...
-    properties.putAll(trinoProperty);
-
-    config.put("properties", properties);
+    config.putAll(trinoProperty);
     return config;
   }
 
+  @Override
+  public Connector buildInternalConnector(Map<String, String> config) throws Exception {
+    return null;
```
Comment (mchades) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java:44]: Always return null?

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoMetadata.java (hunk 7 above)
Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoMetadata.java:180]: `saveMode == SaveMode.IGNORE ` it means `create table if not exist`

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoConfig.java (hunk 5 above)
Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/GravitinoConfig.java:17]: Yes

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java (hunk 2 above)
Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java:426]: I will be used by next pr

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java (hunk 3 above)
Comment (diqiu50) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:46]: It's Depend on the Trino docker environment

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java (hunk 9 above)
Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java:44]: It's not implemented yet

File: build.gradle.kts (hunk 4 above)
Comment (diqiu50) [reply build.gradle.kts:410]: I'm not sure, I remove it first

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java (hunk 3 above)
Comment (mchades) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:46]: But it doesn't use Trino, right?

Comment (diqiu50) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/CatalogsPageTest.java:46]: Yes

**Pull request merged**